### PR TITLE
Subdivision Facegraph

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

The 3D Surface Subdivision Methods package currently requires the input mesh to be of type Polyhedron_3. We generalize the code such that any model of the MutableFaceGraph concept can be used.

The Sqrt3 subdivision was only partially implemented: borders could not be refined. This is corrected, using the technique provided in the original paper that describes the Sqrt3 subdivision.

Improvements and clarification to the User and Reference Manuals are also provided. 

## Release Management

* Affected package(s): Subdivision_method_3
* Feature/Small Feature (if any): [Subdivision FaceGraph](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Subdivision_FaceGraph)

### CODE REVIEW

**Comment by lrineau** [comment]
Was tested last night in CGAL-4.11-Ic-1, and the [documentation results] showed warnings:
```
[...]/doc/Documentation/packages.txt:1041: Warning: unable to resolve reference to `Chapter_3D_Surface_Subdivision_Methods' for \ref command
[...]/doc/Documentation/packages.txt:1041: Warning: unable to resolve reference to `PkgSurfaceSubdivisionMethods3' for \ref command
```
https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.11-Ic-1/Documentation.log

```
Warning: source [...]/Subdivision_method_3/doc/Subdivision_method_3 is not a readable file or directory... skipping.
Warning: source [...]/Subdivision_method_3/include is not a readable file or directory... skipping.
```
https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.11-Ic-1/Subdivision_method_3.log

[documentation results]: https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.11-Ic-1/index.html

**Comment by afabri** [comment]
@lrineau: Successfully tested in [4.11-Ic-4](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.11-Ic-4.shtml)

**Comment by lrineau** [comment]
Compilation error, probably due to 
* 2d889cf - Restructured the subdivision package (2 weeks ago) <Mael Rouxel-Labbé>

```
CGAL-4.11-Ic-13/include/CGAL/Subdivision_method_3/subdivision_hosts_3.h:93:50: error: use of undeclared identifier 'number_of_iterations'; did you mean 'internal_np::number_of_iterations'?
  unsigned int step = choose_param(get_param(np, number_of_iterations), 1);
                                                 ^~~~~~~~~~~~~~~~~~~~
                                                 internal_np::number_of_iterations
/Users/cgaltester/cgal_test/CGAL-4.11-Ic-13/include/CGAL/Polygon_mesh_processing/internal/parameters_interface.h:28:50: note: 'internal_np::number_of_iterations' declared here
CGAL_add_named_parameter(number_of_iterations_t, number_of_iterations, number_of_iterations)
                                                 ^
/Users/cgaltester/cgal_test/CGAL-4.11-Ic-13/include/CGAL/Polygon_mesh_processing/internal/named_function_params.h:38:12: note: expanded from macro 'CGAL_add_named_parameter'
  enum X { Y };
           ^
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.11-Ic-13/Principal_component_analysis_Demo/TestReport_cgaltester_x86-64_Darwin-13.0_Apple-clang-5.0_Release.gz

Actually, it might be a conflict with #1891. I will deselect #1891 for tonight's testsuite, and remerge this one.

**Comment by lrineau** [comment]
... #1891 is already in `master`. You will have to rebase and test locally.

**Comment by MaelRL** [comment]
It's already rebased and fixed in aad0f9f.

File: Subdivision_method_3/include/CGAL/Subdivision_mask_3.h [hunk 1]
```diff
@@ -1,377 +0,0 @@
-// ======================================================================
```
Comment (lrineau) [inline Subdivision_method_3/include/CGAL/Subdivision_mask_3.h]: `<CGAL/Subdivision_mask_3.h>` is a documented header. It cannot be removed.

File: Subdivision_method_3/include/CGAL/Subdivision_method_3.h [hunk 2]
```diff
@@ -1,85 +0,0 @@
-// ======================================================================
```
Comment (lrineau) [inline Subdivision_method_3/include/CGAL/Subdivision_method_3.h]: `<CGAL/Subdivision_method_3.h>` cannot be removed either.

**Comment by MaelRL** [comment]
`<CGAL/Subdivision_mask_3.h>` had to be removed because I added `<CGAL/subdivision_mask_3.h>` to align with typical package structure. Removed the other for one for symmetry but forgot to add this stuff to `changes.html`. Can the depreciation step be skipped in this configuration, provided that I add the breaking changes to `changes.html` ?

**Comment by lrineau** [comment]
Right: that will be similar to the breaking changes of the parameterization package.

**Pull request merged**